src/sdb.c

```diff
@@ -449,7 +449,7 @@ static int sdb_set_internal (Sdb* s, const char *key, char *val, int owned, ut32
 			if (cas && kv->cas != cas) {
 				return 0;
 			}
-			if (vlen == kv->value_len && !strcmp (kv->value, val)) {
+			if (vlen == kv->value_len && strcmp (kv->value, val) != 0) {
```

Comment:
this is wrong. the reason for this check is to ensure that the value is the same, your change swaps the condition so it will not update the value in case its different